signature/algorithm.go

```diff
@@ -15,14 +15,27 @@ type Algorithm int
 //
 // Reference: https://github.com/notaryproject/notaryproject/blob/main/signature-specification.md#algorithm-selection
 const (
-	AlgorithmPS256 Algorithm = 1 + iota // RSASSA-PSS with SHA-256
-	AlgorithmPS384                      // RSASSA-PSS with SHA-384
-	AlgorithmPS512                      // RSASSA-PSS with SHA-512
-	AlgorithmES256                      // ECDSA on secp256r1 with SHA-256
-	AlgorithmES384                      // ECDSA on secp384r1 with SHA-384
-	AlgorithmES512                      // ECDSA on secp521r1 with SHA-512
+	UnsupportedAlgorithm Algorithm = iota
+	AlgorithmPS256                 = iota // RSASSA-PSS with SHA-256
+	AlgorithmPS384                        // RSASSA-PSS with SHA-384
+	AlgorithmPS512                        // RSASSA-PSS with SHA-512
+	AlgorithmES256                        // ECDSA on secp256r1 with SHA-256
+	AlgorithmES384                        // ECDSA on secp384r1 with SHA-384
+	AlgorithmES512                        // ECDSA on secp521r1 with SHA-512
 )
 
+func (alg Algorithm) String() string {
+	return [...]string{
+		"UnsupportedAlgorithm",
+		"RSASSA-PSS-SHA-256",
+		"RSASSA-PSS-SHA-384",
+		"RSASSA-PSS-SHA-512",
+		"ECDSA-secp256r1-SHA-256",
+		"ECDSA-secp384r1-SHA-384",
+		"ECDSA-secp521r1-SHA-512",
+	}[alg]
```

Comment:
Looks like the elements in this string array have to strictly follow the order defined in `const`.  
I've seen an alternative implementation from [go-cose](https://github.com/veraison/go-cose/blob/07c7d28e2c7eb9d6daa4619b98eceeaf7a8e5156/algorithm.go#L52). Using switch statements seems easier to maintain.